### PR TITLE
Searcher bugfix. Closes #232

### LINKED ISSUE #232: Fix bug with searcher matching more than just starting character of first word in TaskEvent name

### PR DESCRIPTION
**Pull request by shaunthium**
Closes #232

### CODE REVIEW

File: src/main/java/cs2103/v15_1j/jimjim/searcher/KeywordFilter.java [hunk 1]
```diff
@@ -20,19 +20,21 @@ public KeywordFilter(List<String> keywords) {
...
-    		for (String word : wordsInName) {
-    			if (word.equalsIgnoreCase(keyword) || 
-    				Character.toLowerCase(word.charAt(0)) == Character.toLowerCase(keyword.charAt(0))) {
-    				found = true;
-    				break;
+    	boolean first = true;
+    	for (String word : wordsInName) {
+    		for (String keyword : keywords) {
+    			if (first) {
+        			if (Character.toLowerCase(word.charAt(0)) == Character.toLowerCase(keyword.charAt(0))) {
```
Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/searcher/KeywordFilter.java]: This still does not fix the problem. The task `milk Shaun` will still show up when Jim searches for `meet`

Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/searcher/KeywordFilter.java]: Better structure:
```
boolean result = false;
if (keyword has 1 character) {
    result = whether the task starts with this character;
}
if (!result) {
    result = check whether the task contain the keyword;
}

File: src/main/java/cs2103/v15_1j/jimjim/searcher/KeywordFilter.java [hunk 2]
```diff
@@ -21,20 +21,20 @@ public KeywordFilter(List<String> keywords) {
...
-    		boolean found = false;
-    		for (String word : wordsInName) {
-    			if (word.equalsIgnoreCase(keyword) || 
-    				Character.toLowerCase(word.charAt(0)) == Character.toLowerCase(keyword.charAt(0))) {
-    				found = true;
-    				break;
+    	for (String word : wordsInName) {
+    		for (String keyword : keywords) {
+    			if (word.equalsIgnoreCase(keyword)) {
+    				return true;
```
Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/searcher/KeywordFilter.java]: just return `word.equalsIgnoreCase(keyword);`

Comment (shaunthium) [inline src/main/java/cs2103/v15_1j/jimjim/searcher/KeywordFilter.java]: That's wrong, I don't want to return false if the first keyword fails. I want to check if the other keywords also match

Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/searcher/KeywordFilter.java]: oops i overlooked this

**Pull request merged**